File: scripts/max_ind.py
import sys

def main():
    if not sys.argv[1:]:
        print("Usage: python %s <input.txt>")
        return 1
    mx = 0
    for line in open(sys.argv[1]):
        toks = line.strip().split()
        ind = int(toks[-1].split(":")[0])
        if ind > mx:
            mx = ind
    sys.stdout.write("Max index: %i\n" % mx)
    return 0

if __name__ == "__main__":
    sys.exit(main())

File: src/max_ind.cpp
#include "train_svm.h"

using std::cerr;

int main(int argc, char *argv[]) {
    if(argc == 1) throw std::runtime_error(std::string("Usage: ") + argv[0] + " <file>");
    int max(0);
    for(char **q(argv); *q; ++q) max = std::max(max, get_max_ind(*q));
    cerr << "Max index: " << max << '\n';
}
